include/dali/dali.h

```diff
@@ -236,8 +239,73 @@ typedef enum _DALIExecType {
    * The dynamic executor offers more flexibility, better memory efficiency and unrestricted
    * lifetime of the pipeline outputs at the expense of more overhead in simple pipelines. */
   DALI_EXEC_DYNAMIC         = DALI_EXEC_ASYNC_PIPELINED | DALI_EXEC_IS_DYNAMIC,
+
+  DALI_EXEC_FORCE_INT32 = 0x7fffffff
 } daliExecType_t;
 
+typedef enum _DALIExecFlags {
+  /** If set, worker threads have CPU affinity set via NVML. */
+  DALI_EXEC_FLAGS_SET_AFFINITY         = 0x00000001,
+
+  /* TODO(michalz): Make concurrency configurable in the pipeline */
+  /** Masks the part of the flags that represent the execution concurrency type. */
+  DALI_EXEC_FLAGS_CONCURRENCY_MASK     = 0x0000000e,
+
+  /** Uses the internally defined default concurrency behavior */
+  DALI_EXEC_FLAGS_CONCURRENCY_DEFAULT  = 0,
+
+  /** Operators are not executed in parallel.
+   *
+   * NOT IMPLEMENTED!
+   * For DALI_EXEC_DYNAMIC only.
+   */
+  DALI_EXEC_FLAGS_CONCURRENCY_NONE     = 1 << 1,
+  /** Operators with different device type (cpu, gpu, mixed) can be executed concurrently.
+   * Operators with the same device are executed sequentially.
+   *
+   * For DALI_EXEC_DYNAMIC only.
+   */
+  DALI_EXEC_FLAGS_CONCURRENCY_BACKEND  = 2 << 1,
+  /** Any two operators may run in parallel.
+   *
+   * NOT IMPLEMENTED!
+   * For DALI_EXEC_DYNAMIC only.
+   */
+  DALI_EXEC_FLAGS_CONCURRENCY_FULL     = 3 << 1,
+
+  /* TODO(michalz): Make stream policy configurable in the pipeline */
+  /** Masks the part of the flags that represent the stream policy. */
+  DALI_EXEC_FLAGS_STREAM_POLICY_MASK   = 0x00000070,
```

Comment:
Do you think we should have some blocks reserved for that? Like every byte?

Reply:
A byte would be too much. I currently reserved 3 bits and now we have 4 values, so we can squeeze 4 more. I can hardly imagine we'd have that many stream policies or concurrency policies.